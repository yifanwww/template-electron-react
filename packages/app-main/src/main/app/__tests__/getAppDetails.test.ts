import { beforeEach, describe, expect, it, jest } from '@jest/globals';

jest.mock('electron', () => {
    const app: Partial<(typeof import('electron'))['app']> = {
        getName: () => 'template-electron-react',
        getVersion: () => '<app-version>',
    };

    return { app };
});

describe('Test fn getAppDetails', () => {
    beforeEach(() => {
        const process = {
            versions: {
                chrome: '<chrome-version>',
                electron: '<electron-version>',
                node: '<node-version>',
                v8: '<v8-version>',
                ares: '<ares-version>',
                http_parser: '<http-parser-version>',
                modules: '<modules-version>',
                openssl: '<openssl-version>',
                uv: '<uv-version>',
                zlib: '<zlib-version>',
            },
        };
        global.process = process as typeof global.process;
    });

    it('should return app details', async () => {
        const { getAppDetails } = await import('../getAppDetails');
        expect(getAppDetails()).toMatchSnapshot();
    });
});
