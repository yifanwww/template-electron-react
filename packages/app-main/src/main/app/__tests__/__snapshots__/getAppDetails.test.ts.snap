// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Test fn getAppDetails should return app details 1`] = `
{
  "module": {
    "chrome": "<chrome-version>",
    "electron": "<electron-version>",
    "node": "<node-version>",
    "v8": "<v8-version>",
  },
  "name": "template-electron-react",
  "version": "<app-version>",
}
`;
